src/index.ts

```diff
@@ -1,3 +1,52 @@
 import "./devices/devices-list";
 import "./components/esphome-header-menu";
 import "./components/esphome-fab";
+import "./editor/esphome-editor";
+import { LitElement, html } from "lit";
+import { customElement, state } from "lit/decorators.js";
+import { editCallback } from "./editor";
+
+@customElement("esphome-main")
+class ESPHomeMainView extends LitElement {
+  @state() private editing?: string;
+  constructor() {
+    super();
+    editCallback.callback = (configuration: string) => {
```

Comment:
What is this? And do we set a global?

Reply:
You already listen to close event on the editor, isn't that the same?

Reply:
if you have a better approach to show up the editor please give more info on how to do that

Reply:
fire an event. That's how we do it in the HA frontend. Events bubble up through the DOM, so you need to make sure you're listening on an element that gets all the events from everywhere, like `document.body`